File: src/java/array/LeftRotation.java
package array;

public class LeftRotation {
    /**
     O(n+m) Time Complexity
     O(n) Space Complexity
     **/

    // Complete the rotLeft function below.
    static int[] rotLeft(int[] a, int d) {
        //grab the size
        int size = a.length;
        int[] rotated_array = new int[size];
        int rotated_index = d;
        int i=0;
        while(rotated_index < size){
            rotated_array[i] = a[rotated_index];
            i++;
            rotated_index++;
        }
        rotated_index = 0;
        while(rotated_index < d){
            rotated_array[i] = a[rotated_index];
            i++;
            rotated_index++;
        }
        return rotated_array;
    }
}

File: classwork/01/arrays/left_rotation.py
# Python Clever
# Watch out for space complexity since 2 additional list copies are being created
def rotLeft(a, d):
    return a[d:] + a[:d]

# O(n) time complexity, O(1) space complexity
def rotLeft(a, d):
    for i in range(d):
        temp = a.pop(0)
        a.append(temp)
    return a
